### PR TITLE
Improve repo initial experience

### PR DESCRIPTION
**Pull request by markwaddle**
- moves dependencies to pyproject.toml instead of requirements.txt.
- adds host and port args
- adds script to start service - `node_engine_service`
- updates instruction
- renames `local_files_root` to `registry_root`

## Motivation and Context
these changes are intended to improve the initial experience with the package:
- dependency management is simpler and more clear (one file, not two)
- service startup is now a script with a reasonable name
- `registry_root` argument more clearly describes what it is for

### CODE REVIEW

File: tools/start-service.sh [hunk 1]
```diff
@@ -6,16 +6,7 @@ set -e
...
-python_cmd=$(command -v python3 &> /dev/null && echo "python3" || echo "python")
-
-# Activate virtual environment
-if [ ! -d "${ROOT}/.venv" ]; then
-    echo "${ROOT}/.venv directory not found"
-    exit 1
-else
-    source ${ROOT}/.venv/bin/activate
-fi
+./tools/activate-venv.sh
```
Comment (dluc) [inline tools/start-service.sh:9]: the file is empty

Comment (markwaddle) [reply tools/start-service.sh:9]: what file is empty? activate-venv.sh? it is not, as far as i can tell.

Comment (markwaddle) [reply tools/start-service.sh:9]: the code in this script was redundant to what activate-venv.sh has, so i replaced it with a call to activate-venv.sh

Comment (markwaddle) [reply tools/start-service.sh:9]: ah! i see it now. it says it is empty, but it is not. i didn't change the contents, only `chmod +x`ed it. it looks like a github bug.

Comment (dluc) [reply tools/start-service.sh:9]: when I click on tools/activate-venv.sh I see an empty file:

image

Comment (dluc) [reply tools/start-service.sh:9]: oh weird, I checked out the PR and I see the content

**Review by dluc — COMMENTED** [review]
LGTM

**Review by dluc — APPROVED** [review]

**Pull request merged**